src/Catty/ViewController/Download/Details/ProjectDetailStoreViewControllerExtension.swift

```diff
@@ -0,0 +1,411 @@
+            counter += 1
+        }
+        setMaxHeightIfGreaterFor(view, withHeight: offset)
+    }
+
+    func addLoadingButton(to view: UIView?, withTarget target: Any?) {
+        let button = EVCircularProgressView()
+        button.tag = Int(kStopLoadingTag)
+        button.tintColor = UIColor.buttonTint
+        let xPosition = (view?.frame.size.width ?? 0.0) - 40
+        let yPosition = (view?.frame.size.height ?? 0.0) * 0.1 + Util.screenHeight() / 4.5
+        button.frame = CGRect(x: xPosition, y: yPosition - 25, width: 28, height: 28)
+        button.isHidden = true
+        button.addTarget(target, action: #selector(URLProtocol.stopLoading), for: .touchUpInside)
+        view?.addSubview(button)
+    }
+
+    func addDownloadAgainButton(to view: UIView?, withTarget target: Any?) {
+        let xPos = (view?.frame.size.width ?? 0.0) / 2 - 10
+        let yPos = (view?.frame.size.height ?? 0.0) * 0.1 + Util.screenHeight() / 4.5
+        let downloadAgainButton = RoundBorderedButton(frame: CGRect(x: xPos, y: yPos - 25, width: 100, height: 25), andBorder: false)
+
+        if let downloadAgainButton = downloadAgainButton {
+            downloadAgainButton.titleLabel?.font = UIFont.boldSystemFont(ofSize: 14)
+            downloadAgainButton.setTitleColor(UIColor.buttonTint, for: .normal)
+            downloadAgainButton.setTitleColor(UIColor.buttonHighlightedTint, for: .highlighted)
+            downloadAgainButton.setTitle( kLocalizedDownload, for: .normal)
+            downloadAgainButton.titleLabel?.adjustsFontSizeToFitWidth = true
+            downloadAgainButton.titleLabel?.minimumScaleFactor = 0.4
+            downloadAgainButton.addTarget(target, action: #selector(downloadAgain), for: .touchUpInside)
+            downloadAgainButton.tag = Int(kDownloadAgainButtonTag)
+            downloadAgainButton.isHidden = true
+            view?.addSubview(downloadAgainButton)
+        }
+    }
+
+    func addOpenButton(to view: UIView?, withTarget target: Any?) {
+        let xPos = (view?.frame.size.width ?? 0.0) - 75
+        let yPos = (view?.frame.size.height ?? 0.0) * 0.1 + Util.screenHeight() / 4.5
+        let openButton = RoundBorderedButton(frame: CGRect(x: xPos, y: yPos - 25, width: 70, height: 25), andBorder: true)
+
+        if let openButton = openButton {
+            openButton.tag = Int(kOpenButtonTag)
+            openButton.isHidden = true
+            openButton.titleLabel?.font = UIFont.boldSystemFont(ofSize: 14)
+            openButton.titleLabel?.adjustsFontSizeToFitWidth = true
+            openButton.titleLabel?.minimumScaleFactor = 0.4
+            openButton.setTitle(kLocalizedOpen, for: .normal)
+            openButton.addTarget(target, action: #selector(openButtonPressed(_:)), for: .touchUpInside)
+            openButton.tintColor = UIColor.buttonTint
+            view?.addSubview(openButton)
+        }
+    }
+
+    func addDownloadButton(to view: UIView?, withTarget target: Any?) {
```

Comment:
The `view` parameter should not be optional. Also, set the scope to `fileprivate`.